lib/utils/qemu.js

```diff
@@ -84,11 +84,12 @@ export function installQemu(arch) {
 				const installStream = fs.createWriteStream(qemuPath);
 
 				const qemuArch = balenaArchToQemuArch(arch);
-				const downloadArchiveName = `qemu-${QEMU_VERSION.replace(
-					/^v/,
-					'',
-				)}-${qemuArch}.tar.gz`;
-				const qemuUrl = `https://github.com/balena-io/qemu/releases/download/${QEMU_VERSION}/${downloadArchiveName}`;
+				const fileVersion = QEMU_VERSION.replace(/^v/, '').replace('+', '.');
+				const urlFile = encodeURIComponent(
+					`qemu-${fileVersion}-${qemuArch}.tar.gz`,
+				);
+				const urlVersion = encodeURIComponent(QEMU_VERSION);
```

Comment:
The new QEMU version string uses a `'+'` character instead of `'-'`:
`'v4.0.0-balena'` _vs._
`'v4.0.0+balena2'`
This requires calling `encodeURIComponent`. Also, the GitHub release filename (download) replaces the `'+'` with a `'.'`. These are the reasons for the minor refactoring.